internal/cmd/base/keyring.go

```diff
@@ -109,24 +112,26 @@ func (c *Command) DiscoverKeyringTokenInfo() (string, string, error) {
 	return keyringType, tokenName, nil
 }
 
-func ReadTokenFromKeyring(ui cli.Ui, keyringType, tokenName string) *authtokens.AuthToken {
+// ReadTokenFromKeyring will attempt to read the token with the name tokenName from the keyring
```

Comment:
Hum... should we consider writing a mock keyring provider (like the pkg does),
so we can write some unit tests?

Reply:
I think this file could do with a refactor to have `DiscoverKeyringTokenInfo` return a keyring provider interface or something, which we could pass around instead of the keyring type, but I think that change is outside the scope of this PR. I've created https://hashicorp.atlassian.net/browse/ICU-16114 to track a future effort to refactor and test this logic.